rockcraft/extensions/go.py

```diff
@@ -0,0 +1,214 @@
+        """Return the parts to add to parts."""
+        return {}
+
+    @override
+    def get_part_snippet(self) -> dict[str, Any]:
+        """Return the part snippet to apply to existing parts."""
+        return {}
+
+    @property
+    def project_name(self) -> str:
+        """Return the normalized name of the rockcraft project."""
+        return self.yaml_data["name"]
+
+    def _check_project(self) -> None:
+        """Check go.mod file exist in project."""
+        if not (self.project_root / "go.mod").exists():
+            raise ExtensionError(
+                "missing go.mod file",
+                doc_slug="/reference/extensions/go-framework",
+                logpath_report=False,
+            )
+
+    def _gen_install_app_part(self) -> Dict[str, Any]:
+        """Generate install-app part with the Go plugin."""
+        install_app = self._get_nested(
+            self.yaml_data, ["parts", "go-framework/install-app"]
+        )
+
+        build_environment = install_app.get("build-environment", [])
+        if self.yaml_data["base"] == "bare":
+            for env_var in build_environment:
+                if "CGO_ENABLED" in env_var:
+                    break
+            else:
+                build_environment = [{"CGO_ENABLED": "0"}]
+
+        organize = install_app.get("organize", {})
+        binary_path = f"usr/local/bin/{self.project_name}"
+        for path in organize.values():
+            if path == binary_path:
+                break
+        else:
+            if not self._get_nested(self.yaml_data, ["services", "go", "command"]):
+                organize[f"bin/{self.project_name}"] = binary_path
+
+        install_app_part = {
+            "plugin": "go",
+            "source": ".",
+            "organize": organize,
+        }
+
+        if not self._check_go_overriden():
+            build_snaps = install_app.get("build-snaps", [])
+            build_snaps.append("go")
+            install_app_part["build-snaps"] = build_snaps
+
+        install_app_part["stage"] = list(organize.values())
+        if build_environment:
+            install_app_part["build-environment"] = build_environment
+
+        return install_app_part
+
+    def _check_go_overriden(self) -> bool:
+        """Check if the user overrode the go snap or package for the build step."""
+        install_app = self._get_nested(
+            self.yaml_data, ["parts", "go-framework/install-app"]
+        )
+        build_snaps = install_app.get("build-snaps", [])
+        if build_snaps:
+            for snap in build_snaps:
+                if snap.startswith("go"):
+                    return True
+        build_packages = install_app.get("build-packages", [])
+        if build_packages:
+            for package in build_packages:
+                if package in ["gccgo-go", "golang-go"]:
+                    return True
+        return False
+
+    def _gen_install_assets_part(self) -> Dict[str, Any] | None:
```

Comment:
```suggestion
    def _get_install_assets_part(self) -> Dict[str, Any] | None:
```

In general I think `gen` at the start of a method implies that it's a generator